packages/core/saved-objects/core-saved-objects-browser-mocks/src/simple_saved_object.mock.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { SavedObjectsClientContract } from '@kbn/core-saved-objects-api-browser';
+import { SimpleSavedObjectImpl } from '@kbn/core-saved-objects-browser-internal';
+import type { SavedObject } from '@kbn/core-saved-objects-common';
+
+const createSimpleSavedObjectMock = (
+  client: SavedObjectsClientContract,
+  savedObject: SavedObject<unknown>
+) => new SimpleSavedObjectImpl(client, savedObject);
```

Comment:
Q: Is it enough to actually create a real instance of `SimpleSavedObjectImpl`? Should we provide an actual mock?

Reply:
@afharo It depends, as usual, on the need vs what we're trying to do. Right now, the main goal is to not have to expose internal methods simply because external consumers are using them for testing, which is what we're doing here.

Answering if we need an actual mock is another story. ATM, there isn't a need (unless I'm missing something).